### PR TITLE
[W5.11][W15-A1]Li yufei

### PR DESCRIPTION
**Pull request by crizyli**
Just ignore the first two commits, they should not be included
two enhancements:
1. add an edit command: edit a person in addressbook
2. make find command case insensitive

### CODE REVIEW

**Comment by nhs-work** [comment]
Good job! The succinct code for making find case insensitive is good! Perhaps it might be good to consider the resources required to find, delete and re-add a person in when using the edit command.

Another point to consider is how editing will cause the edited person's index to be changed. Overall very well done!

File: src/seedu/addressbook/commands/FindCommand.java [hunk 1]
```diff
@@ -50,9 +50,17 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
+            /*if (!Collections.disjoint(wordsInName, keywords)) {
                 matchedPersons.add(person);
+            }*/
+            for (String str1 : wordsInName) {
+                for (String str2 : keywords) {
+                    if(str1.compareToIgnoreCase(str2) == 0 && !matchedPersons.contains(person)) {
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/FindCommand.java:58]: Possible to implement SLAP here.

File: src/seedu/addressbook/commands/FindCommand.java [hunk 2]
```diff
@@ -50,9 +50,17 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
+            /*if (!Collections.disjoint(wordsInName, keywords)) {
                 matchedPersons.add(person);
+            }*/
+            for (String str1 : wordsInName) {
+                for (String str2 : keywords) {
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/FindCommand.java:57]: Nested for-loops should be ignored wherever possible. Try to find ways to efficiently implement this using the existing Collections.disjoint method.

**Review by devamanyu — COMMENTED** [review]
Couldn't see the user guide and test cases update for this enhancement (Find case insensitive). Please include them before next Tutorial.

File: src/seedu/addressbook/commands/EditCommand.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                addressBook.removePerson(toDelete);
+            } else {
+                return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+            }
+            addressBook.addPerson(toEdit);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toEdit));
+        } catch (UniquePersonList.DuplicatePersonException dpe) {
+            return new CommandResult(MESSAGE_DUPLICATE_PERSON);
+        } catch (UniquePersonList.PersonNotFoundException pnfe) {
+            return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/EditCommand.java:82]: Good implementation of the edit person. Duplicacy and missing cases are handled.

**Pull request closed without merging**